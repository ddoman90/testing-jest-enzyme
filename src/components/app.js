import React from 'react';
import CommentBox from './commentBox';
import CommentList from './commentList';

export default props => {
	return (
		<div className="comment-box">
			<h4>Add comment</h4>
			<CommentBox />
			<CommentList />
		</div>
	);
};
