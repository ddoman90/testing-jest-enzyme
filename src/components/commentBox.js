import React, { Component } from 'react';

const INITIAL_STATE = {
	comment: ''
};

export default class CommentBox extends Component {
	constructor(props) {
		super(props);

		this.state = INITIAL_STATE;
	}
	onChange({ target }) {
		this.setState({
			comment: target.value
		});
	}

	onSubmit(event) {
		event.preventDefault();
		this.setState(INITIAL_STATE);
	}

	render() {
		return (
			<form className="comment_box" onSubmit={this.onSubmit.bind(this)}>
				<textarea
					className="comment_text"
					name="comment"
					value={this.state.comment}
					onChange={this.onChange.bind(this)}
				/>
				<button action="submit">Submit</button>
			</form>
		);
	}
}
