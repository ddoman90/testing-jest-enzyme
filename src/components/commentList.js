import React, { Component } from 'react';
import { connect } from 'react-redux';

export class CommentList extends Component {
	render() {
		const comments = this.props.comments || [];
		return (
			<div className="comment_list">
				{comments.map(comment => <li key={comment}>{comment}</li>)}
			</div>
		);
	}
}

export default connect(({ comments }) => {
	return { comments };
})(CommentList);
