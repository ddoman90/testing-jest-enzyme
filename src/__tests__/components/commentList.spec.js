import React from 'react';
import { shallow } from 'enzyme';
// import the dumb component for unit test
import { CommentList } from '../../components/commentList';
import renderer from 'react-test-renderer';

describe('CommentList', () => {
	let component;

	beforeEach(() => {
		const comments = ['new comment', 'other new comment'];
		component = shallow(<CommentList comments={comments} />);
	});

	it('renders as intended', () => {
		const tree = renderer.create(<CommentList />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders each li', () => {
		expect(component.find('li').length).toBe(2);
	});

	it('renders each li with appropriate text', () => {
		expect(component.text()).toMatch('new comment');
		expect(component.text()).toMatch('other new comment');
	});
});
