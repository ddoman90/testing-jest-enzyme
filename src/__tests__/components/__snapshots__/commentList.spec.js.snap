// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList renders as intended 1`] = `
<div
  className="comment_list"
/>
`;
